container_restart_without_docker.md

```diff
@@ -0,0 +1,133 @@
+in the image, or its ancestors. 4. Any command or args elements in the k8s yaml
+(described
+[here](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes)).
+5. Any environment variables defined in the k8s yaml, and any secrets or
+configmaps referenced by those environment variables (described
+[here](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#use-environment-variables-to-define-arguments)).
+
+Note also that CMD and ENTRYPOINT can be specified in `shell` (plain string) or
+`exec` (array of strings) mode, and we'd want to preserve that distinction.
+
+I have seen no public library functions that will perform any of this logic for
+us, which leaves us implementing it ourselves (including stuff like: base image
+has `ENV FOO=1`, child image has `ENV BAR=${FOO}2`, `ENTRYPOINT ${BAR}`).
+
+Aside from the technical risk of getting those details right, all of the
+behaviors described above are for a specific version of Docker and Kubernetes.
+I do not know if they have changed in the past or will change in the future. If
+a user runs Tilt with a version of Docker or Kubernetes that has different
+behavior, we're now potentially running the wrong command for them.
+
+While it doesn't necessarily make sense to rule out ever implementing this
+logic, it does seem pretty expensive and risky to do for what currently seems
+to be a niche case for which there is an existing workaround.
+
+Another option would be to implement only a subset of the logic, if we could
+write guards to detect and give good error messages on cases we don't yet
+support (e.g., environment variables). I'm not sure writing those guards is
+significantly cheaper than implementing the proper logic itself, but it reduces
+exposure to versioning risks (which I'm not sure are significant).
+
+#### Run the image and observe the entry point
+We could quickly run the container, grab argv for its process 1, and throw it
+away. This seems fraught:
+1. If the process has side effects, this is bad.
+2. It seems confusing to the user for this ephemeral pod/container to show up.
+3. We'd probably want to do this in a way that's as hidden from the user as
+   possible, like in a pod that doesn't have a manifest associated with it
+   so that its status / logs don't show up in the UI.
+4. If the code is in a state such that the process crashes on startup then we
+   can't get its argv this way (AFAIK) and we have a new failure mode that's
+   kinda awkward to explain to users.
+   1. Well, "process crashes on startup" is really an existing failure mode,
+      except now we're doing it in a case where the failure log isn't streamed,
+      and we're not quite using the existing pod watch flow.
+
+#### Require the user to specify the entrypoint
+OK, so we need the entrypoint but it's hard to get. Why don't we just have the
+user tell us what entrypoint to use?
```

Comment:
i think giving a syntax example here (while noting that naming is TBD) would help ppl wrap their heads around this. (better name here v much appreciated)
```docker_build(...,
    live_update[
        ...
        rerun_container_process('/go/bin/my_app')
    ]
)
```
also, worth making a note here (just so WE remember) that we need to decide how this does/doesn't interact with entrypointOverride?